src/Config.h

```diff
@@ -0,0 +1,1488 @@
+
+/**
+ * @file Config.h
+ *
+ * @brief Centralized configuration management with JSON storage
+ */
+
+#pragma once
+
+#include "Logger.h"
+#include "defaults.h"
+#include "hardware/Relay.h"
+#include "hardware/Switch.h"
+#include <ArduinoJson.h>
+#include <LittleFS.h>
+
+class Config {
+    public:
+        /**
+         * @brief Initialize the configuration system
+         *
+         * @return true if successful, false otherwise
+         */
+        bool begin() {
+            if (!LittleFS.begin(true)) {
+                LOG(ERROR, "Failed to initialize LittleFS");
+                return false;
+            }
+
+            // Check if config file exists
+            if (!LittleFS.exists(CONFIG_FILE)) {
+                LOG(INFO, "Config file not found, creating from defaults");
+                createDefaults();
+
+                return save();
+            }
+
+            // Try to load existing config
+            if (!load()) {
+                LOG(WARNING, "Failed to load config, creating from defaults");
+                createDefaults();
+
+                return save();
+            }
+
+            return true;
+        }
+
+        /**
+         * @brief Create a new configuration with default values
+         */
+        void createDefaults() {
```

Comment:
this is only called by `begin()` and should not be available to any other class dealing with a `Config` object, make it `private` (and possibly rename to `create_defaults()` to indicate the "private-ness".